### PR TITLE
Issue 36 cpc2 download

### LINKED ISSUE #36: CPC2 download is erratic
Two possible solutions here:

1. Use a different mirror from the official, unreliable one. I have cloned the code myself here ago, but this repo seems slightly more up to date:

https://github.com/biocoder/CPC2/

2. Potentially find a different source for this?

### PR DESCRIPTION
**Pull request by lucventurini**
Closes #36 by using the GitHub clone.

### CODE REVIEW

File: minos/scripts/install_cpc2.py [hunk 1]
```diff
@@ -30,27 +30,28 @@ def main():
 	args.install_path = os.path.abspath(args.install_path)
 	
 	with cd(args.install_path):
-		cmd = "wget http://cpc2.cbi.pku.edu.cn/data/CPC2-beta.tar.gz"
+		cmd = "wget --no-check-certificate --content-disposition https://github.com/biocoder/CPC2/tarball/9a2b596"
```
Comment (cschu) [inline minos/scripts/install_cpc2.py:33]: maybe add the hash as a command line arg to the install script. updates are not likely, but i could imagine submitting the py3 patch as a PR to the biocoda repo.

**Review by cschu — COMMENTED** [review]
looks good, only one suggestion - s. below

File: minos/scripts/install_cpc2.py (hunk 1 above)
Comment (lucventurini) [reply minos/scripts/install_cpc2.py]: Done

File: minos/scripts/install_cpc2.py [hunk 2]
```diff
@@ -24,34 +24,39 @@ def cd(newdir):
 def main():
 
 	ap = argparse.ArgumentParser()
-	ap.add_argument("install_path", default=".", type=str)
+	ap.add_argument("--hash", default="9a2b596", help="Hash or version tag to use for downloading CPC2 from github")
+	ap.add_argument("--repo-name", default="CPC2", help="Name of the repository in GitHub to download from.")
+	ap.add_argument("-u", "--username", default="biocoder", help="Username associated with the CPC2 repo on GitHub")
+	ap.add_argument("install_path", default=".", type=str, nargs="?")
```
Comment (cschu) [inline minos/scripts/install_cpc2.py:30]: even better!

File: minos/scripts/install_cpc2.py [hunk 3]
```diff
@@ -24,34 +24,39 @@ def cd(newdir):
...
-		cmd = "wget --no-check-certificate --content-disposition https://github.com/biocoder/CPC2/tarball/9a2b596"
+		os.makedirs("CPC2", exist_ok=True)
+		cmd = "wget -O CPC2.tar.gz --no-check-certificate --content-disposition https://github.com/{args.username}/{args.repo_name}/tarball/{args.hash}".format(
+			args=args)
 		print("Downloading CPC2...", end="", flush=True)
-		p = subprocess.check_output(cmd, shell=True, stderr=subprocess.STDOUT)
+		subprocess.check_output(cmd, shell=True, stderr=subprocess.STDOUT)
 		print(" done.")
-		cmd = "tar xvzf biocoder-CPC2-9a2b596.tar.gz"
+		cmd = "tar --strip-components=1 -C CPC2 -xvzf CPC2.tar.gz".format(args=args)
```
Comment (cschu) [inline minos/scripts/install_cpc2.py:43]: why is there a `.format()`? I don't see any 'placeholders' in the cmd string.

File: minos/scripts/install_cpc2.py [hunk 4]
```diff
@@ -24,34 +24,39 @@ def cd(newdir):
...
 		print("Unpacking...", end="", flush=True)
-		p = subprocess.check_output(cmd, shell=True, stderr=subprocess.STDOUT)
+		subprocess.check_output(cmd, shell=True, stderr=subprocess.STDOUT)
 		print(" done.")
 
-		cmd = "rm -f biocoder-CPC2-9a2b596.tar.gz"
+		cmd = "rm -f CPC2.tar.gz".format(args=args)
```
Comment (cschu) [inline minos/scripts/install_cpc2.py:48]: s. above

File: minos/scripts/install_cpc2.py [hunk 5]
```diff
@@ -24,34 +24,39 @@ def cd(newdir):
...
 		print("Removing tarball...", end="", flush=True)
-		p = subprocess.check_output(cmd, shell=True, stderr=subprocess.STDOUT)
+		subprocess.check_output(cmd, shell=True, stderr=subprocess.STDOUT)
 		print(" done.")
 
-		cmd = "cd biocoder-CPC2-9a2b596/libs/libsvm/libsvm-3.18 && make clean && make"
+		cmd = "cd CPC2/libs/libsvm/libsvm-3.18 && make clean && make".format(args=args)
```
Comment (cschu) [inline minos/scripts/install_cpc2.py:53]: s. above

**Review by cschu — COMMENTED** [review]
yea, that's good - just minor questions

File: minos/scripts/install_cpc2.py (hunk 5 above)
Comment (lucventurini) [reply minos/scripts/install_cpc2.py:53]: Ah. I did put it there while I was still having the CPC2 folder taking the name from the username and the hash ...

**Pull request merged**